iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/task/subtask/connector/PipeRealtimePriorityBlockingQueue.java

```diff
@@ -71,30 +78,53 @@ public boolean put(final Event event) {
 
   @Override
   public Event directPoll() {
-    Event event = super.directPoll();
+    Event event = null;
+    if (eventCount.get() >= realTimeEventConsumeThreshold) {
```

Comment:
There're some slightly non-atomic issues.. Two subtasks may concurrently call the "pollFirst", breaking the strictness of the parameter. May use updateAndGet....